### PR TITLE
Exporters-intro rework and mention benefits of OTLP exporters

### LINKED ISSUE #3295: Update exporters intro with advantages of OTLP
If there is a refactor in the future, consider adding something along the lines of:

1. OTLP Exporters are built to carry OTel data model without any loss of information. (Other exporters may have info loss, like some exporters may not have a good place to store resources, instrumentation scope etc. but OTLP carry everything).
2. OTLP is the only exporter that supports all 3 signals (logs/metrics/traces).

_Originally posted by @cijothomas in https://github.com/open-telemetry/opentelemetry.io/pull/3276#discussion_r1330205694_

### PR DESCRIPTION
**Pull request by chalin**
- Proposes a fix #3295
- Closes #3902 - sorry @pegasas it was easier to start from a clean slate than comment on your PR.
- If @open-telemetry/docs-approvers agree with this approach, I'll double check if I need to clean up other language Exporter pages (I've only tackled two so far in this PR).

**Preview**:

- https://deploy-preview-3905--opentelemetry.netlify.app/docs/languages/java/exporters/
- https://deploy-preview-3905--opentelemetry.netlify.app/docs/languages/go/exporters/

### Screenshot (sample)

> image

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]

File: layouts/shortcodes/docs/languages/exporters-intro.md [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
+## Available exporters
+
+The registry contains the [list of exporters for {{ $name }}][reg].
 
-As part of {{ $name }} you will find many exporters being available. Among them,
-the OpenTelemetry Protocol (OTLP) exporters provide the best experience for you
-as an end-user, since it is a general-purpose telemetry data delivery protocol
-designed in the scope of the OpenTelemetry project.
+Among exporters, [OpenTelemetry Protocol (OTLP)][OTLP] exporters provide an
+optimal experience since they are lossless and support all [signals]. To learn
```
Comment (cijothomas) [inline layouts/shortcodes/docs/languages/exporters-intro.md:21]: "lossless" needs some clarification... It could be misinterpreted as "100% reliable".

Comment (chalin) [reply layouts/shortcodes/docs/languages/exporters-intro.md:21]: Hmm, you're right. Ok, I'll try to think of something better, but it'll likely have to wait a couple of days, I need to shift projects for a bit.

If someone has other suggested wording, suggestions are most welcome. (@cartermp has some good ideas :))

Comment (cijothomas) [reply layouts/shortcodes/docs/languages/exporters-intro.md:21]: OTLP Exporters are designed with OTel data model (link to it) in mind, allowing it to carry OTel data model without any loss of information. (Other exporters may have information loss, like some exporters may not have a good place to store resources, instrumentation scope etc. but OTLP carry everything).

Comment (cartermp) [reply layouts/shortcodes/docs/languages/exporters-intro.md:21]: Here's a proposed phrasing based largely on @cijothomas's words:

> Among exporters, [OpenTelemetry Protocol (OTLP)][OTLP] exporters are
designed to carry OpenTelemetry data without any loss of information. Furthermore,
there is an abundance of tools that operate on telemetry and support OTLP,
allowing you to a high degree of flexibility when you need it. To learn
more about the OTLP protocol, see [OTLP Specification][OTLP].

Comment (cijothomas) [reply layouts/shortcodes/docs/languages/exporters-intro.md:21]: You can also call out that, some tools like [Jaeger ](https://opentelemetry.io/blog/2022/jaeger-native-otlp/) and [Prometheus ](https://prometheus.io/docs/prometheus/latest/feature_flags/#otlp-receiver) now natively accept OTLP.

Comment (svrnm) [reply layouts/shortcodes/docs/languages/exporters-intro.md:21]: I think that's a very good point, not only jaeger and prometheus but all the vendors on our vendor page.

Comment (chalin) [reply layouts/shortcodes/docs/languages/exporters-intro.md]: I've incorporated all of your feedback and revisited the paragraph text. PTAL.

File: layouts/shortcodes/docs/languages/exporters-intro.md [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
 
-{{ $lang := .Get 0 -}} {{ $data := index $.Site.Data.instrumentation $lang }}
-{{ $name := cond (isset $.Site.Data.instrumentation $lang) (printf "OpenTelemetry %s" $data.name) "the language specific implementations" -}}
+{{ $data := index $.Site.Data.instrumentation $lang -}}
 
-In order to visualize and analyze your telemetry, you will need to export your
-data to an [OpenTelemetry Collector](/docs/collector/) or a backend such as
+{{ $name := $data.name -}}
+
+To visualize and analyze telemetry, export it to the
```
Comment (cijothomas) [inline layouts/shortcodes/docs/languages/exporters-intro.md:7]: if we want to be precise - then we should revisit this... This gives the impression that Collector is a  tool for "visualizing and analyzing", but it is not. Not introduced in this PR, so not a blocker for this.

**Review by cijothomas — APPROVED** [review]

File: layouts/shortcodes/docs/languages/exporters-intro.md (hunk 2 above)
Comment (chalin) [reply layouts/shortcodes/docs/languages/exporters-intro.md:7]: Let's track this via:

- #3951

Thanks for the review and approval @cijothomas.

**Comment by svrnm** [comment]
@chalin , apologies merging the PR to update the short codes on the exporter pages created that conflict.

File: layouts/shortcodes/docs/languages/exporters-intro.md [hunk 3]
```diff
@@ -1,22 +1,34 @@
-<!-- cSpell:ignore isset cond -->
+{{ $lang := .Get 0 -}}
 
-{{ $lang := .Get 0 -}} {{ $data := index $.Site.Data.instrumentation $lang }}
-{{ $name := cond (isset $.Site.Data.instrumentation $lang) (printf "OpenTelemetry %s" $data.name) "the language specific implementations" -}}
+{{ $data := index $.Site.Data.instrumentation $lang -}}
```
Comment (svrnm) [inline layouts/shortcodes/docs/languages/exporters-intro.md:3]: We need to keep this, this is used on the concept pages about exporters

**Comment by chalin** [comment]
@svrnm - no worries. I've rebased, resolved conflicts, and slightly reworked the initial part of the exporters intro. PTAL

**Review by svrnm — APPROVED** [review]

**Pull request merged**